app/src/scenes/Badges/BadgeModal.js

```diff
@@ -33,13 +33,24 @@ import { logEvent } from '../../services/logEventsWithMatomo';
     secondaryButtonLink: null,
 }
 */
-
 const BadgeModal = () => {
   const navigation = useNavigation();
   const [showModal, setShowModal] = useState(false);
   const [modalContent, setModalContent] = useState(null);
   const setBadges = useSetRecoilState(badgesState);
   const setBadgesCatalog = useSetRecoilState(badgesCatalogState);
+  const [lastBadge, setLastBadge] = useState(false);
+  const isLastBadge = () => {
```

Comment:
https://react.dev/learn/you-might-not-need-an-effect
https://react.dev/reference/react/useMemo#skipping-re-rendering-of-components

<img width="1300" alt="image" src="https://user-images.githubusercontent.com/31724752/228260334-ece81c35-6a71-49dd-ab4a-dc26729ed9fb.png">


```
const isLastBadge = useMemo(() => {
    if (modalContent?.category === 'articles') {
      if (modalContent?.stars === 4) return true;
      return false;
    } 
     if (modalContent?.stars === 5 && !modalContent?.category.includes('locked')) {
          return true;
    }
    return false;
  }, [modalContent]);
  ```